### PR TITLE
Added new method to unsubscribe from all channels and groups with completino block

### PR DESCRIPTION
**Pull request by parfeon**
Now it client has additional method `-unsubscribeFromAllWithCompletion:` which will call passed completion block after `leave` REST API will receive response from service.

### CODE REVIEW

File: Example/PubNub Example.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -501,7 +501,9 @@
 				CLANG_ENABLE_MODULES = YES;
 				CLANG_ENABLE_OBJC_ARC = YES;
 				CLANG_WARN_ASSIGN_ENUM = YES;
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
```
Comment (jzucker2) [inline Example/PubNub Example.xcodeproj/project.pbxproj]: Why these changes to the .pbxproj file?

File: Framework/PubNub Framework.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -5070,6 +5076,8 @@
 				CLANG_WARN_INFINITE_RECURSION = YES;
 				CLANG_WARN_INT_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
+				CLANG_WARN_STRICT_PROTOTYPES = YES;
```
Comment (jzucker2) [inline Framework/PubNub Framework.xcodeproj/project.pbxproj]: Same question for these changes too

File: Tests/PubNub Tests.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1416,6 +1416,7 @@
 		519C32771B20BF4D00FAC283 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
```
Comment (jzucker2) [inline Tests/PubNub Tests.xcodeproj/project.pbxproj]: And these changes

File: Example/PubNub Example.xcodeproj/project.pbxproj (hunk 1 above)
Comment (parfeon) [reply Example/PubNub Example.xcodeproj/project.pbxproj]: Well, I've tried to apply suggested configuration, partly managed, but Xcode still suggest to change a lot about build process and architectures.

File: Framework/PubNub Framework.xcodeproj/project.pbxproj (hunk 2 above)
Comment (parfeon) [reply Framework/PubNub Framework.xcodeproj/project.pbxproj]: Same as above :)

File: Tests/PubNub Tests.xcodeproj/project.pbxproj (hunk 3 above)
Comment (parfeon) [reply Tests/PubNub Tests.xcodeproj/project.pbxproj]: Same as above of above :D

**Comment by jzucker2** [comment]
OK, will wait for CI to run before I approve. Looks great to me
